backend/src/main/java/com/rota/auth/AuthenticationUtils.java

```diff
@@ -1,51 +1,207 @@
+
   /**
-   * Checks if the provided token is valid.
-   * Only returns false if token is "false" (case insensitive).
-   * @param token Authentication token.
-   * @return Whether authentication was successful.
+   * TODO decide if it is definitely necessary to check with Auth0. Another method for that perhaps?
+   * Checks if the current token is valid locally and via Auth0.
+   *
+   * @return Whether the current token is valid.
    */
-  public static boolean validateToken(String token) {
-    // This will only return false if the token is "false" (case insensitive)
-    return !token.toLowerCase().equals("false");
+  public boolean validateToken(String token) {
+    HttpRequest request = HttpRequest.newBuilder()
+        .GET()
+        .uri(URI.create("https://rota-flex-101.eu.auth0.com/userinfo"))
+        .setHeader("Authorization", "Bearer " + token)
+        .build();
+
+    HttpResponse<String> response;
+    try {
+      response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+    } catch (IOException | InterruptedException e) {
+      // This should hopefully never happen
+      throw new AuthenticationHttpException(e);
+    }
+    return SecurityContextHolder.getContext().getAuthentication().isAuthenticated()
+        && response.statusCode() == 200;
   }
 
   /**
-   * Extracts user information from the provided token.
-   * @param token Authentication token.
-   * @return User information if token is valid and if present.
+   * Finds the user in the database using their email.
+   *
+   * @param token Valid authentication token.
+   * @return {@link Staff} member with that email.
    */
-  public static Optional<Integer> getUserFromToken(String token) {
-    Integer parsedToken;
-    try {
-      parsedToken = Integer.parseInt(token);
-    } catch (NumberFormatException e) {
-      return Optional.empty();
-    }
-    return Optional.of(parsedToken);
+  public Staff getUserFromToken(String token) {
+    JsonNode userInfo = getUserJsonFromToken(token);
+    return staffService.findStaffByEmail(userInfo.get("email").asText());
   }
 
   /**
    * TODO Temp solution needs to be changed to be more appropriate when auth is added to backend.
    * Get user role from token.
+   *
    * @param token Authentication token
    * @return {@link Role} of the user or <code>null</code> if token is not valid.
    */
-  public static Optional<Role> getUserRoleFromToken(String token) {
+  public Optional<Role> getUserRoleFromToken(String token) {
     return validateToken(token)
         ? Optional.of(
-          (token.length() % 2 == 0)
+        (token.length() % 2 == 0)
             ? Role.MANAGER
             : Role.USER
-          )
+    )
         : Optional.empty();
   }
+
+  /**
+   * Uses the provided access token to get user information from Auth0.
+   *
+   * @param token Valid authentication token.
+   * @return JsonNode of Auth0 information
+   */
+  public JsonNode getUserJsonFromToken(String token) {
```

Comment:
can this be made private?

Reply:
We need this method to get the actual userInfo json, and then use the other utility methods to get bits of information from the json currently.